### PR TITLE
[Release test] [Cluster launcher] Add release test for aws `example-full.yaml`

### PR DESCRIPTION
**Pull request by architkulkarni**
## Why are these changes needed?
Adds a release test for `example-full.yaml` on AWS.

Starts the cluster with `ray up`, runs a simple Ray driver script, and calls `ray down`.

Also fixes a bug in this YAML file where we were using a string instead of an int for a VolumeSize.

## Checks

- [ ] I've signed off every commit(by using the -s flag, i.e., `git commit -s`) in this PR.
- [ ] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
    - [ ] I've added any new APIs to the API Reference. For example, if I added a 
           method in Tune, I've added it in `doc/source/tune/api/` under the 
           corresponding `.rst` file.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [ ] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by architkulkarni** [comment]
New test passed: https://buildkite.com/ray-project/release-tests-pr/builds/35317#0187912c-0d01-4683-a08a-4611140bd151

Screenshot 2023-04-17 at 4 45 21 PM

File: python/ray/autoscaler/aws/example-full.yaml [hunk 1]
```diff
@@ -79,7 +79,7 @@ available_node_types:
             BlockDeviceMappings:
                 - DeviceName: /dev/sda1
                   Ebs:
-                      VolumeSize: 140GB
+                      VolumeSize: 140
```
Comment (architkulkarni) [inline python/ray/autoscaler/aws/example-full.yaml:82]: This used to fail with
```python-traceback
Traceback (most recent call last):
  File "/home/ray/anaconda3/bin/ray", line 8, in <module>
    sys.exit(main())
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/scripts/scripts.py", line 2448, in main
    return cli()
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1130, in __call__
    return self.main(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1055, in main
    rv = self.invoke(ctx)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1657, in invoke
    return _process_result(sub_ctx.command.invoke(sub_ctx))
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 1404, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/click/core.py", line 760, in invoke
    return __callback(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/autoscaler/_private/cli_logger.py", line 856, in wrapper
    return f(*args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/scripts/scripts.py", line 1272, in up
    use_login_shells=use_login_shells,
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/autoscaler/_private/commands.py", line 289, in create_or_update_cluster
    no_monitor_on_head,
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/autoscaler/_private/commands.py", line 726, in get_or_create_head_node
    provider.create_node(head_node_config, head_node_tags, 1)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/autoscaler/_private/aws/node_provider.py", line 320, in create_node
    created_nodes_dict = self._create_node(node_config, tags, count)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/ray/autoscaler/_private/aws/node_provider.py", line 424, in _create_node
    created = self.ec2_fail_fast.create_instances(**conf)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/boto3/resources/factory.py", line 580, in do_action
    response = action(self, *args, **kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/boto3/resources/action.py", line 88, in __call__
    response = getattr(parent.meta.client, operation_name)(*args, **params)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/botocore/client.py", line 530, in _api_call
    return self._make_api_call(operation_name, kwargs)
  File "/home/ray/anaconda3/lib/python3.7/site-packages/botocore/client.py", line 924, in _make_api_call
    headers=additional_headers,
  File "/home/ray/anaconda3/lib/python3.7/site-packages/botocore/client.py", line 991, in _convert_to_request_dict
    api_params, operation_model
  File "/home/ray/anaconda3/lib/python3.7/site-packages/botocore/validate.py", line 381, in serialize_to_request
    raise ParamValidationError(report=report.generate_report())
botocore.exceptions.ParamValidationError: Parameter validation failed:
Invalid type for parameter BlockDeviceMappings[0].Ebs.VolumeSize, value: 140GB, type: <class 'str'>, valid types: <class 'int'>
```

**Comment by wuisawesome** [comment]
Can you post an example of a green e2e run when this is ready?

**Comment by architkulkarni** [comment]
> Can you post an example of a green e2e run when this is ready?

https://github.com/ray-project/ray/pull/34487#issuecomment-1512243576

**Review by wuisawesome — APPROVED** [review]

**Comment by architkulkarni** [comment]
Lint passed. Other failed tests unrelated, this PR just adds a release test.

**Pull request merged**